### PR TITLE
Fix some ARG002 per file ignores

### PR DESCRIPTION
**Pull request by MaximSmolskiy**
### Describe your change:

Fix some `ARG002` per file ignores

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: audio_filters/show_response.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 
 class FilterType(Protocol):
-    def process(self, sample: float) -> float:
+    def process(self, sample: float) -> float:  # noqa: ARG002
```
Comment (cclauss) [inline audio_filters/show_response.py:11]: What would happen if we delete the unused variables?

Comment (MaximSmolskiy) [reply audio_filters/show_response.py]: This class is used as interface, so I made this method absctract. Now there are no problems with unused method argument in this place

File: data_structures/hashing/hash_table.py [hunk 2]
```diff
@@ -173,7 +173,7 @@ def _set_value(self, key, data):
         self.values[key] = data
         self._keys[key] = data
 
-    def _collision_resolution(self, key, data=None):
+    def _collision_resolution(self, key, data=None):  # noqa: ARG002
```
Comment (cclauss) [inline data_structures/hashing/hash_table.py:176]: Same here?

Comment (MaximSmolskiy) [reply data_structures/hashing/hash_table.py]: Yes. Done. `noqa` left in `data_structures/hashing/quadratic_probing.py` where this method is overridden

**Review by cclauss — APPROVED** [review]
OK!

We get these warnings towards the end of the GItHub build test run...
```
web_programming/instagram_crawler.py .                                   [ 99%]
/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/site-packages/coverage/report_core.py:115: CoverageWarning: Couldn't parse '/home/runner/work/Python/Python/config-3.py': No source for code: '/home/runner/work/Python/Python/config-3.py'. (couldnt-parse)
  coverage._warn(msg, slug="couldnt-parse")
/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/site-packages/coverage/report_core.py:115: CoverageWarning: Couldn't parse '/home/runner/work/Python/Python/config.py': No source for code: '/home/runner/work/Python/Python/config.py'. (couldnt-parse)
  coverage._warn(msg, slug="couldnt-parse")
web_programming/test_fetch_github_info.py .                              [100%]
=============================== warnings summary ===============================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.MessageMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.ScalarMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.
linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py::linear_algebra.src.gaussian_elimination_pivoting.gaussian_elimination_pivoting.solve_linear_system
  /home/runner/work/Python/Python/linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:55: RuntimeWarning: invalid value encountered in scalar divide
    ab[i, column_num - 1]
sorts/odd_even_transposition_parallel.py: 45 warnings
  /opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=1839) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

**Pull request merged**